File: java-sdk/src/main/java/io/fyno/core_java/utils/FynoUtils.java
package io.fyno.core_java.utils;

public class FynoUtils {
    public String getEndpoint(String event, String ws, String env, String profile, String newId, String version) {
        String baseEndpoint = FynoConstants.PROD_ENDPOINT;

        if (env != null && env.equals("test")) {
            baseEndpoint = FynoConstants.DEV_ENDPOINT;
        }

        String commonPath = ws + "/" + version + "/" + FynoConstants.PROFILE;

        switch (event) {
            case "create_profile":
                return baseEndpoint + "/" + commonPath;
            case "get_profile":
            case "upsert_profile":
                return baseEndpoint + "/" + commonPath + "/" + profile;
            case "merge_profile":
                return baseEndpoint + "/" + commonPath + "/" + profile + "/merge/" + newId;
            case "update_channel":
                return baseEndpoint + "/" + commonPath + "/" + profile + "/channel";
            case "delete_channel":
                return baseEndpoint + "/" + commonPath + "/" + profile + "/channel/delete";
            case "event_trigger":
                return baseEndpoint + "/" + ws + "/" + version + "/" + FynoConstants.EVENT_PATH;
            default:
                return "";
        }
    }
}

File: fyno-kotlin-core/src/main/java/io/fyno/core/utils/FynoUtils.kt
package io.fyno.core.utils

import android.util.Log
import io.fyno.core.FynoCore

class FynoUtils {
    fun getEndpoint(
        event: String,
        ws: String,
        env: String? = "live",
        profile: String? = null,
        newId: String? = null,
        version: String? = "live"
    ): String {
        val baseEndpoint =
            if (env == "test") FynoConstants.DEV_ENDPOINT else FynoConstants.PROD_ENDPOINT
        val commonPath = "$ws/track/$version/${FynoConstants.PROFILE}"
        Logger.i(
            "GET_ENDPOINT",
            "getEndpoint Params: $event, $ws, $env, $profile, $newId, $version"
        )

        var url = when (event) {
            "create_profile" -> "$baseEndpoint/$commonPath"
            "merge_profile" -> "$baseEndpoint/$commonPath/$profile/merge/$newId?appVersion=${
                FynoCore.getString(
                    "appVersionName"
                )
            }&package=${FynoCore.getString("appPackageName")}"

            "upsert_profile" -> "$baseEndpoint/$commonPath/$profile?appVersion=${
                FynoCore.getString(
                    "appVersionName"
                )
            }&package=${FynoCore.getString("appPackageName")}\""

            "update_channel" -> "$baseEndpoint/$commonPath/$profile/channel?appVersion=${
                FynoCore.getString(
                    "appVersionName"
                )
            }&package=${FynoCore.getString("appPackageName")}\""

            "delete_channel" -> "$baseEndpoint/$commonPath/$profile/channel/delete?appVersion=${
                FynoCore.getString(
                    "appVersionName"
                )
            }&package=${FynoCore.getString("appPackageName")}\""

            else -> ""
        }

        return url
    }
}